Breadcrumbs.py

```diff
@@ -32,80 +38,169 @@ def get_row_indentation(points, view, tab_size, limit=1e20):
+  if is_white_row(view, get_points_by_row(current_row)):
+    while 0 <= current_row and is_white_row(view, get_points_by_row(current_row)):
       current_row -= 1
 
-    breadcrumbs = []
-    last_line_start = view.text_point(current_row + 1, 0)
-    while 0 <= current_row and 0 < indentation:
-      line_start = get_row_start(current_row)
-      points = xrange(line_start, last_line_start)
-      last_line_start = line_start
-      current_indentation = get_row_indentation(points, view, tab_size, indentation)
-      if current_indentation < indentation and not is_white_row(view, points):
-        indentation = current_indentation
-        breadcrumbs.append(get_breadcrumb(view, points, breadcrumb_length_limit))
+    if current_row < 0:
+      return
 
-      current_row -= 1
+    indentation = get_row_indentation(get_points_by_row(current_row), view, tab_size) + 1
+  else:
+    indentation = get_row_indentation(get_points_by_row(current_row), view, tab_size)
+    current_row -= 1
+
+  breadcrumbs = []
+  last_line_start = view.text_point(current_row + 1, 0)
+  while 0 <= current_row and 0 < indentation:
+    line_start = get_row_start(current_row)
+    points = xrange(line_start, last_line_start)
+    last_line_start = line_start
+    current_indentation = get_row_indentation(points, view, tab_size, indentation)
+    if current_indentation < indentation and not is_white_row(view, points):
+      indentation = current_indentation
+      breadcrumbs.append(get_breadcrumb(view, points, my_regex, separator))
+
+    current_row -= 1
+
+  breadcrumbs.reverse()
+  lengths = [len(breadcrumb) for breadcrumb in breadcrumbs]
+  sorted_lengths = sorted(lengths)
+  previous_length = 0
+  number_of_characters_left  = total_breadcrumbs_length_limit - len(lengths) * len(separator)
+  for (number_of_shorter, current_length) in enumerate(sorted_lengths):
+    if previous_length < current_length:
+      previous_length = current_length
+      number_of_not_shorter = len(breadcrumbs) - number_of_shorter
+      if number_of_characters_left < current_length * number_of_not_shorter:
+        length_of_short_trim = number_of_characters_left // number_of_not_shorter
+        length_of_long_trim = length_of_short_trim + 1
+        number_of_long_trimmed = number_of_characters_left % number_of_not_shorter
+        number_of_short_trimmed = number_of_not_shorter - number_of_long_trimmed
+        for (index_of_breadcrumb, breadcrum_length) in enumerate(lengths):
+          if current_length <= breadcrum_length:
+            if 0 < number_of_short_trimmed:
+              length_of_trim = length_of_short_trim
+              number_of_short_trimmed -= 1
+            else:
+              length_of_trim = length_of_long_trim
+            breadcrumbs[index_of_breadcrumb] = breadcrumbs[index_of_breadcrumb][0:length_of_trim]
+        break
+    number_of_characters_left -= current_length
+
+  return ''.join(breadcrumbs+[''])
```

Comment:
> As stated elsewhere, I'd rather return the breadcrumbs array here, and let the caller of this function decide what to do with it.

Good idea.

> As pointed elsewhere, to make the arithmetic correct etc. it would be better to use separator to join the breadcrumbs.

Done.

> but I do not know if I can trust this documentation

I think it may be lagging behind a bit. wbond seems to be super busy polishing everything non-code for the next public release, so I expect the docs to get updated later on.